src/backend/main/go.main/facilities/REST/contacts.go

```diff
@@ -279,6 +281,41 @@ func (rest *RESTfacility) ContactExists(userID, contactID string) bool {
 	return rest.store.ContactExists(userID, contactID)
 }
 
+// Process a vcard file and create related contacts
+func (rest *RESTfacility) ImportVcardFile(info *UserInfo, file io.Reader) error {
+	vcards, err := contact.ParseVcardFile(file)
+	if err != nil {
+		return err
+	}
+	log.Debug("[ImportVcardFile] Have parse ", len(vcards), " vcards")
+
+	importErrors := make([]error, 0, len(vcards))
+	for _, card := range vcards {
+		contact, err := contact.FromVcard(info, card)
+		if err != nil {
+			log.Warn("[ImportVcardFile] Error during vcard transformation ", err)
+			importErrors = append(importErrors, err)
+		} else {
+			err = rest.CreateContact(info, contact)
+			if err != nil {
+				log.Warn("[ImportVcardFile] Create contact failed with error ", err)
```

Comment:
`rest.CreateContact` reports which uri is blocking within this contact in case of duplication with another contact. We could add a TODO here to handle this specific error and try to remove conflicting uri before calling again `rest.CreateContact`. It would improve the number of imported contacts.

Reply:
These array of error will need further processing and first to be reported to user.

For the moment we reject all conflicting vcard